### PR TITLE
Coverity

### PR DESCRIPTION
**Pull request by yschaeff**

### CODE REVIEW

**Comment by yschaeff** [comment]
#build

File: signer/src/signer/zone.c [hunk 1]
```diff
@@ -58,7 +58,7 @@ zone_create(char* name, ldns_rr_class klass)
     if (!name || !klass) {
         return NULL;
     }
-    CHECKALLOC(zone = (zone_type*) malloc(sizeof(zone_type)));
+    CHECKALLOC(zone = (zone_type*) calloc(1, sizeof(zone_type)));
```
Comment (halderen) [inline signer/src/signer/zone.c]: Here you are really just masking a problem, namely that the zone_type isn't really fully initialized when it returns.  The initialization of ends should be moved/passed through this place.  It still remains uninitialized (but guaranteed to be NULL), but the code does not expect this entry not to be null....

**Comment by yschaeff** [comment]
#build
These are all the high priority issues (as classified by coverity)

**Comment by yschaeff** [comment]
final #build for this branch. Will not add any more commits if it succeeds. In that case feel free to review and merge.

**Comment by yschaeff** [comment]
#build

**Pull request merged**